src/rendezvous_addr.rs

```diff
@@ -198,3 +160,70 @@ pub fn rendezvous_addr(
         })
         .into_boxed()
 }
+
```

Comment:
If you want to factor all this out it would probably be clearer to make it into its own object which implements `Future` rather than making a function which 5 `&mut` parameters then gets wrapped in a `future::poll_fn`

Reply:
Right, I guess. Well, that's the first step of the process. This already allows me to test smaller, more isolated code. I don't wanna spend too much time on this though. But since I did this while investigating `rendezvous_addr()`, I thought I could also merge this to the mainstream branch since it brings some value.
What do you think: should I completely refactor it, should I put some TODO notes or should I leave it as it is? :)

Reply:
I think just refactor it. It won't be that hard, it'll just be something like:

```rust
struct RendezvousAddrStun {
    active_queries: .. ,
    ports: ..,
    errors: ..,
    known_ip_opt: ..,
    igd_error: ..,
    failed_sequences: ..,
}

impl Future for RendezvousAddrStun {
    type Item = SocketAddr;
    type Error = RendezvousAddrError;

    fn poll(&mut self) -> .. {
        // that function body goes here
    }
}
```

then stick `RendezvousAddrStun { /* fields */ }` where we currently call `poll_fn`.

Reply:
ok

Reply:
You can remove the `poll_fn` here entirely because it's not doing anything. Just replace it with the `PublicAddrsFromStun`. Also, is there any reason why the `PublicAddrsFromStun` takes a bunch of `&mut` references? It would be simpler if it just took ownership.

Reply:
hmm, what about servers polling? see line 103

Reply:
I would move all that into the object as well. You can have separate methods on the object for polling the active queries and for polling the incoming servers.